vehicle/mkvi/software/imu/icm20948/icm20948.h

```diff
@@ -0,0 +1,256 @@
+#define GYRO_CONFIG_2      0x02
+#define XG_OFFS_USRH       0x03
+#define XG_OFFS_USRL       0x04
+#define YG_OFFS_USRH       0x05
+#define YG_OFFS_USRL       0x06
+#define ZG_OFFS_USRH       0x07
+#define ZG_OFFS_USRL       0x08
+#define ODR_ALIGN_EN       0x09
+#define ACCEL_SMPLRT_DIV_1 0x10
+#define ACCEL_SMPLRT_DIV_2 0x11
+#define ACCEL_INTEL_CRTL   0x12
+#define ACCEL_WOM_THR      0x13
+#define ACCEL_CONFIG       0x14
+#define ACCEL_CONFIG_2     0x15
+#define FSYNC_CONFIG       0x52
+#define TEMP_CONFIG        0x53
+#define MOD_CTRL_USR       0x54
+// #define REG_BANK_SEL 0x7F
+
+// Bank 3
+#define I2C_MST_ODR_CONFIG 0x00
+#define I2C_MST_CTRL       0x01
+#define I2C_MST_DELAY_CTRL 0x02
+#define I2C_SLV0_ADDR      0x03
+#define I2C_SLV0_REG       0x04
+#define I2C_SLV0_CTRL      0x05
+#define I2C_SLV0_DO        0x06
+#define I2C_SLV1_ADDR      0x07
+#define I2C_SLV1_REG       0x08
+#define I2C_SLV1_CTRL      0x09
+#define I2C_SLV1_DO        0x0A
+#define I2C_SLV2_ADDR      0x0B
+#define I2C_SLV2_REG       0x0C
+#define I2C_SLV2_CTRL      0x0D
+#define I2C_SLV2_DO        0x0E
+#define I2C_SLV3_ADDR      0x0F
+#define I2C_SLV3_REG       0x10
+#define I2C_SLV3_CTRL      0x11
+#define I2C_SLV3_DO        0x12
+#define I2C_SLV4_ADDR      0x13
+#define I2C_SLV4_REG       0x14
+#define I2C_SLV4_CTRL      0x15
+#define I2C_SLV4_DO        0x16
+#define I2C_SLV4_DI        0x17
+#define REG_BANK_SEL       0x7F
+
+// Magnetometer I2C address
+#define MAGNETOMETER_ADDR 0x0C
+
+/**
+ * Magnetometer register map
+ */
+#define WIA2  0x01
+#define ST1   0x10
+#define HXL   0x11
+#define HXH   0x12
+#define HYL   0x13
+#define HYH   0x14
+#define HZL   0x15
+#define HZH   0x16
+#define ST2   0x18
+#define CNTL2 0x31
+#define CNTL3 0x32
+#define TS1   0x33
+#define TS2   0x34
+
+/**
+ * Bit for setting read/write status in I2C address.
+ */
+#define I2C_READ  1
+#define I2C_WRITE 0
+
+/**
+ * Initialize the internal I2C controlled magnetometer in the ICM.
+ */
+void init_magnetometer(void);
+
+/**
+ * Read a register in the magnetometer.
+ *
+ * @param register_ (uint8_t) The register to read.
+ * @param rx_data (uint8_t*) A pointer to store the read data.
+ */
+void read_magnetometer(uint8_t register_, uint8_t* rx_data);
+
+/**
+ * Write a register in the magnetometer.
+ *
+ * @param register_ (uint8_t) The register to write.
+ * @param tx_data (uint8_t) The data to write.
+ */
+void write_magnetometer(uint8_t _register, uint8_t tx_data);
+
+/**
+ * Read data from an ICM register.
+ *
+ * @param _register (uint8_t) The register to read.
+ * @param rx_data (uint8_t*) A pointer to the read data.
+ */
+void icm_read_register(uint8_t _register, uint8_t* rx_data);
+
+/**
+ * Write data to an ICM register.
+ *
+ * @param _register (uint8_t) The register to write.
+ * @param tx_data (uint8_t) The data to write.
+ */
+void icm_write_register(uint8_t _register, uint8_t tx_data);
+
+/**
+ * ICM register banks.
+ */
+#define BANK_0 0
+#define BANK_1 1
```

Comment:
nit: make this an enum?

Reply:
Added a todo. Don't have the hardware so I don't want to change too much without testing.